### ISSUE TITLE
Create a forked angular preboot with preboot_nonce support

### ISSUE THREAD

**Issue by rquast** [aurelia/ssr-engine#9]
ssr-engine currently uses the angular preboot directly. This is an issue for the following reasons:

- Aurelia isn't in the angular team's scope of work. They could introduce breaking changes.
- PREBOOT_NONCE support for CSP written in their preboot code is only for Angular. There needs to be an Aurelia version.
- Keeping documentation in sync is difficult because changes in angular preboot don't get updated in aurelia's ssr code.

Desired action:

- Fork the project and implement preboot_nonce as part of the work (doesn't look difficult).
- Create a schedule to check for additions/bug fixes in angular preboot. I don't know if there is an existing system for this in Aurelia's workflow system. How are dependencies currently checked for updates in Aurelia?

**Comment by EisenbergEffect** [comment]
I agree that we should fork and do work to customize for Aurelia. Regarding the schedule, we don't really have any dependencies for Aurelia's core. We do for i18n. Perhaps @zew666 can comment on how he has handled that.

**Comment by zewa666** [comment]
Im not sure I fully understand what the question with regards to i18n is @Eisenbergeffect. I havent tried using the Plugin in the serverside so this could impose a todo.

**Comment by EisenbergEffect** [comment]
Sorry! The question is how have you kept up with changes in the dependencies to i18n? Did you have any particular way of managing that?

**Comment by zewa666** [comment]
Oh I was very bad at that as i've just waited for the first user to ask for a v bump. Don't take me as a good example ;)

**Comment by zewa666** [comment]
So I have thought about that a bit and things like https://greenkeeper.io/ or the free https://renovateapp.com/ github extension might be a way to get automatically updated for changes of the origin repo. That would help on a PR level.

As for version changes having a custom script firing `npm outdated PACKAGENAME` could give us info whether we're on track or changes happened.

**Comment by fkleuver** [comment]
I don't know too much about github/javascript-specific CI as I'm more of a Visual Studio / Azure guy, but if you go for a 3rd party service like Greenkeeper you'll want to really commit to it. Or it's just a wsted effort. So gGrab unit testing too while you're at it, and do it across more projects. 

I think if we're just talking pure dependency version management - there are not that many in the aurelia libraries - a simple hand rolled script that's run every now and then might be plenty. The important thing is you shouldn't have to remember more than the name of the script to run.

**Comment by zewa666** [comment]
@fkleuver I'm personally also leaning more towards a simple script which runs and just notifies you about an outdated dependency. The issue here of course is that somebody then always has to look up what changed and manually merge the changes. But it's certainly good enough to start with.

As for remembering, I think thats not really needed since this script could be called as part of our CircleCI integration and be defined either on a warning level, to continue although outdated, or error level to break hard.

**Comment by rquast** [comment]
@zewa666 I like the idea with CircleCI checking the versions. However, in the case of angular preboot where it will no longer be a dependency since it is forked, how do you automatically check for updates there?

**Comment by rquast** [comment]
@fkleuver I think webpack changes are a good example of where this affects Aurelia. There are a lot of dev dependencies that break things. Who is in charge of updating the CLI to migrate to webpack 4 since it's only working with 3 at the moment?

**Comment by EisenbergEffect** [comment]
In the specific case of the CLI, we've got it updated for 4.0 but we haven't released yet because we want to make Webpack the default CLI setting and to do that we need to update our tutorial documentation. I've almost finished fixing that up, so that release should go out in the next day or so :) 

For Webpack specifically, several members of our team stay on top of what's happening in the Webpack community. So, when there's a big change, they try to help update our plugins and then we try to get that coordinated with any other tools. This particular upgrade is taking a bit longer due to the reasons above. I'm a bit behind schedule due to some life stuff this week as well. It will be sorted soon.

**Comment by wzrdtales** [comment]
Forking it will make sense, especially since in the latest version they have direct dependencies to angular.
